invisible_cities/detsim/simulate_s1_test.py

```diff
@@ -0,0 +1,175 @@
+import os
+import numpy as np
+
+from .. io.dst_io  import load_dst
+
+from . simulate_s1 import compute_scintillation_photons
+from . simulate_s1 import compute_s1_pes_at_pmts
+from . simulate_s1 import generate_s1_time
+from . simulate_s1 import generate_s1_times_from_pes
+from . simulate_s1 import s1_waveforms_creator
+from . simulate_s1 import histogram_s1_times
+from . simulate_s1 import create_lighttable_function
+from . simulate_s1 import binedges_from_bincenters
+
+from pytest import raises
+
+from hypothesis import given, settings
+from hypothesis.strategies  import integers
+from hypothesis.strategies  import floats
+from hypothesis.extra.numpy import arrays
+
+
+def test_compute_scintillation_photons():
+    np.random.seed(1234)
+    energy = np.array([1, 2, 3, 4, 5])
+    ws = 1.
+    expected = np.array([0, 4, 5, 5, 5])
+    np.testing.assert_allclose(expected, compute_scintillation_photons(energy, ws))
+
+
+def test_compute_s1_pes_at_pmts(ICDATADIR):
+    s1_lighttable = os.path.join(ICDATADIR, "NEXT_NEW.energy.S1.PmtR11410.LightTable.h5")
+    S1_LT = create_lighttable_function(s1_lighttable)
+
+    xs = np.array([1, 2]); ys = np.array([1, 2]); zs = np.array([1, 2])
+    photons = np.array([1000, 1000])
+    s1_pes_at_pmts = compute_s1_pes_at_pmts(xs, ys, zs, photons, S1_LT)
+
+    assert s1_pes_at_pmts.shape == (12, len(xs))
+
+
+def test_generate_s1_time():
+    np.random.seed(1234)
+    times = generate_s1_time(size=2)
+    expected = np.array([164.77783654,  46.96530216])
+
+    np.testing.assert_allclose(times, expected)
+
+
+def test_generate_s1_times_from_pes():
+    np.random.seed(1234)
+    s1_pes_at_pmts = np.array([[1, 0, 1],
+                               [0, 2, 1]])  # 2 pmts, 3 hits
+    hit_times = np.array([0, 1, 2])
+    expected  = [np.array([164.77783654,  48.96530216]),
+                 np.array([83.11706209, 24.66602705, 26.35333038])]
+
+    values = generate_s1_times_from_pes(s1_pes_at_pmts, hit_times)
+
+    assert len(values) == len(expected)
+    for exp, val in zip(expected, values):
+        np.testing.assert_allclose(exp, val)
+
+
+def test_histogram_s1_times():
+    s1_times = np.array([[1, 2, 3, 4, 5]])
+    buffer_length = 5
+    bin_width     = 1
+    start_time    = 1
+
+    expected = np.array([[1, 1, 1, 1, 1]])
+    h = histogram_s1_times(s1_times, buffer_length, bin_width, start_time)
+
+    np.testing.assert_allclose(h, expected)
+
+
+def find_nearest_(array, value):
+    """Finds nearest element of an array"""
+    idx = (np.abs(array - value)).argmin()
+    return array[idx]
+
+@settings(max_examples=500)
+@given(xs = arrays(np.float, 1, elements=floats(min_value=-500, max_value=500)),
+       ys = arrays(np.float, 1, elements=floats(min_value=-500, max_value=500)),
+       zs = arrays(np.float, 1, elements=floats(min_value=-500, max_value=500)))
+def test_get_lt_values_s1(ICDATADIR, xs, ys, zs):
+
+    s1_lighttable = os.path.join(ICDATADIR, "NEXT_NEW.energy.S1.PmtR11410.LightTable.h5")
```

Comment:
This should be a fixture (something like e15d2063ac415291c68454695791233cd22523b9 , you can probably copy the code and add S1 table as well)